pv_site_api/main.py

```diff
@@ -1,448 +1,23 @@
+from .auth import Auth
+from .config import Config
+from .session import create_session_dependency
 
-import pandas as pd
-import sentry_sdk
-import structlog
-from dotenv import load_dotenv
-from fastapi import Depends, FastAPI, HTTPException
-from fastapi.middleware.cors import CORSMiddleware
-from fastapi.openapi.utils import get_openapi
-from fastapi.responses import FileResponse, JSONResponse
-from pvlib import irradiance, location, pvsystem
-from pvsite_datamodel.read.site import get_all_sites, get_site_by_uuid
-from pvsite_datamodel.read.status import get_latest_status
-from pvsite_datamodel.sqlmodels import ClientSQL, SiteSQL
-from pvsite_datamodel.write.generation import insert_generation_values
-from sqlalchemy.orm import Session
+config = Config()
 
-import pv_site_api
-
-from ._db_helpers import (
-    does_site_exist,
-    get_forecasts_by_sites,
-    get_generation_by_sites,
-    site_to_pydantic,
-)
-from .fake import (
-    fake_site_uuid,
-    make_fake_forecast,
-    make_fake_pv_generation,
-    make_fake_site,
-    make_fake_status,
-)
-from .pydantic_models import (
-    ClearskyEstimate,
-    Forecast,
-    MultiplePVActual,
-    PVSiteAPIStatus,
-    PVSiteMetadata,
-    PVSites,
+get_session = create_session_dependency(
+    db_url=config.DB_URL,
+    is_fake=config.FAKE,
 )
-from .redoc_theme import get_redoc_html_with_theme
-from .session import get_session
-from .utils import get_yesterday_midnight
-
-load_dotenv()
-
-logger = structlog.stdlib.get_logger()
-
-
-def traces_sampler(sampling_context):
-    """
-    Filter tracing for sentry logs.
-
-    Examine provided context data (including parent decision, if any)
-    along with anything in the global namespace to compute the sample rate
-    or sampling decision for this transaction
-    """
-
-    if os.getenv("ENVIRONMENT") == "local":
-        return 0.0
-    elif "error" in sampling_context["transaction_context"]["name"]:
-        # These are important - take a big sample
-        return 1.0
-    elif sampling_context["parent_sampled"] is True:
-        # These aren't something worth tracking - drop all transactions like this
-        return 0.0
-    else:
-        # Default sample rate
-        return 0.05
-
-
-def is_fake():
-    return int(os.environ.get("FAKE", 0))
-
 
-sentry_sdk.init(
-    dsn=os.getenv("SENTRY_DSN"),
-    environment=os.getenv("ENVIRONMENT", "local"),
-    traces_sampler=traces_sampler,
+auth = Auth(
+    domain=config.AUTH0_DOMAIN,
+    api_audience=config.AUTH0_API_AUDIENCE,
+    algorithm=config.AUTH0_ALGORITHM,
 )
 
-app = FastAPI(docs_url="/swagger", redoc_url=None)
-
-title = "Nowcasting PV Site API"
-
-folder = os.path.dirname(os.path.abspath(__file__))
-description = """
-Description of PV Site API
-"""
-
-origins = os.getenv("ORIGINS", "*").split(",")
-app.add_middleware(
-    CORSMiddleware,
-    allow_origins=origins,
-    allow_credentials=True,
-    allow_methods=["*"],
-    allow_headers=["*"],
+app = create_app(
```

Comment:
Now the entry point is cleaner: we only define the dependencies and inject them into the `create_app` function.